### PR TITLE
Task 22: FE: Aside filter component. Implement brands filter

### PR DESCRIPTION
**Pull request by di-kinosian**
Снимок экрана (45)

### CODE REVIEW

File: src/components/Filters/AsideFilter.tsx [hunk 1]
```diff
@@ -1,12 +1,47 @@
...
+import { url } from "../../main/constants/common";
 
 export const AsideFilter = () => {
+  const [priceRange, setPriceRange] = useState<PriceRange>({
+    min: 0,
+    max: 1000,
+  });
+  const [availableRange, setAvailableRange] = useState<PriceRange>({
+    min: 0,
+    max: 1000,
```
Comment (VYaniv) [inline src/components/Filters/AsideFilter.tsx:17]: It has to be dynamic. Code duplicate can be in the separate constant

File: src/components/Filters/AsideFilter.tsx [hunk 2]
```diff
@@ -1,12 +1,47 @@
...
+  const [brands, setBrands] = useState<string[]>([]);
+
+  useEffect(() => {
+    const getFilters = async () => {
+      try {
+        const response = await axios.get(`${url}/products/filters`);
+        setAvailableRange(response.data.price);
+        setPriceRange(response.data.price);
+        setBrands(response.data.brands);
+      } catch (error) {
```
Comment (VYaniv) [inline src/components/Filters/AsideFilter.tsx:28]: I recommend integrating all filters into one state. Even using form

File: src/components/Filters/PriceFilter.tsx [hunk 3]
```diff
@@ -1,56 +1,92 @@
...
+  }, [value]);
+
+  const handleSliderChange = (_: Event, newValue: number | number[]): void => {
+    const [min, max] = newValue as number[];
+    onChange({ min, max });
+    setMinInputValue(min.toString());
+    setMaxInputValue(max.toString());
+  };
+
+  const onMaxInputBlur = (e: FocusEvent<HTMLInputElement>) => {
```
Comment (VYaniv) [inline src/components/Filters/PriceFilter.tsx:34]: Returned type

File: src/components/Filters/AsideFilter.tsx (hunk 1 above)
Comment (di-kinosian) [reply src/components/Filters/AsideFilter.tsx:17]: done

**Review by VYaniv — APPROVED** [review]

**Pull request merged**